libraries/resource_providers/image_streamer/api300/deployment_plan.rb

```diff
@@ -0,0 +1,43 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+require_relative '../../../resource_provider'
+
+module OneviewCookbook
+  module ImageStreamer
+    module API300
+      # DeploymentPlan Provider resource methods
+      class DeploymentPlanProvider < ResourceProvider
+        # Generic method to retrieve and return the URI from different image streamer resources using the name of the resource
+        # @param [String, Symbol] i3s_resource_type Type of image streamer resource to be retrieved. e.g., ':GoldenImage'.
+        # @param [String] i3s_resource_name Name of the resource from context.
+        def load_i3s_resource(i3s_resource_type, i3s_resource_name)
```

Comment:
I'd usually like to have a raise in case a resource I'm passing in does not exist, but if it's all the same, I can drop the method and just swap out the method calls for the aforementioned `data['myTypeUri'] = resource_named(:MyType).find_by(name: @context.my_type) if @context.my_type`